### ISSUE TITLE
Lint/FormatParameterMismatch triggers off format strings in arguments (ie strftime arguments)

### ISSUE THREAD

**Issue by seandilda** [rubocop-hq/rubocop#8115]
Lint/FormatParameterMismatch appears to trigger if the arguments include a string that appears to have format values in it.

--------

## Expected behavior

The cop should only complain if there are conflicting parameter types in the format string.

## Actual behavior

If the format string has one type of parameter types and an argument to the format string appears to have another, the cop will trigger.

## Steps to reproduce the problem

Run rubocop against a file with the following line:

```ruby
format('%<time>s', time: time.strftime('%Y-%m-%d %H:%M:%S'))
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
0.85.1 (using Parser 2.7.1.3, rubocop-ast 0.0.3, running on ruby 2.6.6 x86_64-linu
x-musl)
```

**Comment by bbatsov** [comment]
@andrykonchin can you look into this, please? I'm assuming it's related to the recent changes you did there.

**Comment by andrykonchin** [comment]
@bbatsov Done
